### DISCUSSION TITLE
How to determine the source coordinates better in computer vision speed estimation

### DISCUSSION THREAD

**Discussion by dearMOMO** [roboflow/supervision#1335] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

blog    https://blog.roboflow.com/estimate-speed-computer-vision/

![image](https://github.com/roboflow/supervision/assets/34231503/62335a5a-5133-4c22-be2a-4ee76ce5f4d6)

Is there A good tool to calculate the coordinates of A, B, C, D? Thank you very much

**Comment by likith1908** [comment]
@dearMOMO did you find any solution to this?

**Comment by aliyilma** [comment]
Good question, let me diversify the topic a bit according to my own work. Is there an easy method for perspective correction/determination? In my own work, sometimes I need to do perspective correction of the video, sometimes I just need to transform for object traces. It can be done somehow with OpenCV but it complicates the code and I don't get very good results. It would be nice if there was an easy method in the Supervision library. For example, detection/trajectories could be transformed according to the rectangle drawn on the screen and given its dimensions, and even projected in a second image window (this could be useful for other annotation methods like tracks). 

There is a very nice automated implementation in the case study, especially for dynamic camera angles.
[https://www.youtube.com/watch?v=yJWAtr3kvPU](https://www.youtube.com/watch?v=yJWAtr3kvPU)
The tracker software also has an interactive perspective conversion tool for video and coordinates
[https://physlets.org/tracker/help/video_filters.html](https://physlets.org/tracker/help/video_filters.html)

<img width="801" alt="image" src="https://github.com/user-attachments/assets/3db58adb-0fbb-41b8-9845-952446bc74f8">

<img width="400" alt="image" src="https://physlets.org/tracker/help_images/perspective_filter_input.gif"><img width="400" alt="image" src="https://physlets.org/tracker/help_images/perspective_filter_output.gif">

**Reply by SkalskiP** [reply]
Hi 👋🏻 I used keypoint detection models to do it. 

https://github.com/user-attachments/assets/0b801dd5-baa7-49c1-858b-259e702b1d17

**Reply by aliyilma** [reply]
hi, awesome work! looked at your code [()](https://github.com/roboflow/notebooks/blob/main/notebooks/how-to-track-football-players.ipynb) and couldn't find how to project players to the 2D field, or it might be because my coding skills not very good. in my case, i am working on different crowded scenes and keypoint detection might not be useful with those.

**Reply by SkalskiP** [reply]
@aliyilma [this](https://github.com/roboflow/notebooks/blob/main/notebooks/how-to-track-football-players.ipynb) is my old demo project; the code for this project will be hosted [here](https://github.com/roboflow/sports) soon.
